### PR TITLE
dotCMS/core#27468 fix and restore JSON signature of generateText

### PR DESCRIPTION
**Pull request by dsilvam**
* Fixed and restored the version of `$ai.generateText` which takes a Map.  This is important to have since it can take custom config values instead of the ones set at portlet level
* Made the `$ai.generateImage` methods public. Both were private. 
* Send back to user the actual error messages / proper handing of the "error" responses on the ImageService.

### CODE REVIEW

File: src/main/java/com/dotcms/ai/service/OpenAIChatService.java [hunk 1]
```diff
@@ -5,4 +5,6 @@
 public interface OpenAIChatService {
 
     JSONObject sendTextPrompt(String prompt);
+
+    JSONObject sendRawRequest(JSONObject prompt);
```
Comment (jdotcms) [inline src/main/java/com/dotcms/ai/service/OpenAIChatService.java:36]: doc

File: src/main/java/com/dotcms/ai/service/OpenAIChatServiceImpl.java [hunk 2]
```diff
@@ -17,18 +17,27 @@ public OpenAIChatServiceImpl(AppConfig appConfig) {
...
-                    Map.of("role","user", "content", prompt)
+                    Map.of("role","user", "content", prompt.getString("prompt"))
             );
-            request.put("messages", messages);
-
+            prompt.put("messages", messages);
         }
-        return new JSONObject(OpenAIRequest.doRequest(config.getApiUrl(),"POST",config.getApiKey(),request)) ;
+
+        prompt.remove("prompt");
```
Comment (jdotcms) [inline src/main/java/com/dotcms/ai/service/OpenAIChatServiceImpl.java:40]: Modify the parameter is usually a smell code, think about it

**Review by wezell — APPROVED** [review]
Some doc would be good.

File: src/main/java/com/dotcms/ai/service/OpenAIChatService.java (hunk 1 above)
Comment (dsilvam) [reply src/main/java/com/dotcms/ai/service/OpenAIChatService.java:36]: Done

**Review by freddyDOTCMS — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**